lsr_fingerprint.py

```diff
@@ -44,12 +47,40 @@
                         ]
                         for tmpl in tmpls:
                             tmplpath = join(dirpath, tmpl)
-                            with open(tmplpath) as fp:
-                                lines = fp.read()
-                            newlines = lines.replace(
-                                "{0}:{1}".format(role["reponame"], role["rolename"]),
-                                "system_role:{0}".format(role["lsrrolename"]),
-                            )
-                            if lines != newlines:
-                                with open(tmplpath, "w") as fp:
-                                    fp.write(newlines)
+                            with open(tmplpath) as ifp:
+                                lines = ifp.read()
+                            if lines.find(oldpair):
+                                with open(tmplpath) as ifp:
+                                    lines = ifp.readlines()
+                                count = 0
+                                newlines = ""
+                                changed = {}
+                                for line in lines:
+                                    newline = line.replace(oldpair, newpair)
+                                    if newline != line:
+                                        if len(changed) > 0:
+                                            raise Exception(
+                                                "Error: duplicate fingerprint '{0}' found in {1}:{2}".format(
+                                                    line.strip(),
+                                                    tmplpath,
+                                                    count,
+                                                )
+                                            )
+                                        changed = {
+                                            "path": tmplpath,
+                                            "linenumber": count,
+                                            "oldline": line.strip(),
+                                            "newline": newline.strip(),
+                                        }
+                                        changedlist.append(changed)
+                                    newlines = "{0}{1}".format(newlines, newline)
+                                    count += 1
+                                with open(tmplpath, "w") as ofp:
+                                    ofp.writelines(newlines)
```

Comment:
`writelines` is typically used with a `list` or other iterable - I guess it would work with a `str` type, but it's confusing that that argument to `writelines` is a `str`.  Could you do
```
56: newlines = []
...
76: newlines.append(newline)
...
ofp.writelines(newlines)
```